Presentation/Program.cs

```diff
@@ -0,0 +1,500 @@
+using Application.Users.GetUserByUsernameAndPassword;
+using Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using MediatR;
+// temporary
+using Domain.Exceptions;
+using System.Text.Json;
+using Domain;
+using System.Reflection;
+using System.Net.NetworkInformation;
+using Application.Users.AddUsers;
+using Application.Users.AddUser;
+using Application.Messages.AddMessages;
+using Application.Messages.GetMessagesBetweenTwoUsers;
+using Application.Users.RemoveUser;
+using Application.Users.GetAllDisplayNames;
+using Application.Users.GetUserById;
+using Application.Users.ValidateIdFriend;
+using Application.Users.AcceptOrRemoveFriendRequest;
+using Application.Users.SendFriendRequest;
+using Application.Users.RemoveFriend;
+using Application.Users.GetUsersCount;
+using Application.Messages.CheckIfMessageValid;
+using Application.Messages.AddMessage;
+using Application.Messages.GetMessages;
+using Application.Users.GetUsers;
+// to here
+
+namespace Presentation
+{
+    public class Program
+    {
+        public static User Login(IMediator mediator)
+        {
+            Console.Write("username: ");
+            string? username = Console.ReadLine();
+            Console.Write("password: ");
+            string? password = Console.ReadLine();
+
+            try
+            {
+                User user = mediator.Send(new GetUserByUsernameAndPassword
+                {
+                    Username = username,
+                    Password = password
+                }).Result;
+
+                return user;
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        public static void LoginMenu(IMediator mediator)
+        {
+            User loggedUser;
+
+            while (true)
+            {
+                try
+                {
+                    loggedUser = Login(mediator);
+
+                    Console.Clear();
+
+                    mediator.Send(new RemoveUser
+                    {
+                        User = loggedUser
+                    });
+
+                    PreFriendsMenu(loggedUser, mediator);
+
+                    mediator.Send(new AddUser
+                    {
+                        User = loggedUser
+                    });
+                }
+                catch (AggregateException ex)
+                {
+                    Console.WriteLine(ex.InnerException.Message);
+                    break;
+                }
+            }
+        }
+
+        public static int ConvertInputToInt(string choiceString)
+        {
+            int receiverIndex;
+            if (int.TryParse(choiceString, out receiverIndex) == false)
+            {
+                return -1;
+            }
+            return receiverIndex;
+        }
+
+        public static bool CheckIfChoiceValid(int choice, int length)
+        {
+            if (choice <= 0 || choice > length)
+            {
+                throw new NumberBetweenException(length);
```

Comment:
In future, you can add validation instead of throwing an exception.
https://docs.fluentvalidation.net/en/latest/aspnet.html

Reply:
I am throwing exceptions because, when I attended the exception handling presentation, Norbert (the guy that was presenting) told us to use exceptions in our project and I thought it'd be a good idea to throw exceptions when the user's input is incorrect. When I told Norbert about it, he said it was ok, so I rolled with it.